File: workspace/computational intelligence/src/differentialEvolution/Genotype.java
package differentialEvolution;

public class Genotype {
	final int NVARS = 2;
	//final int MAXGENS = 10000;
	final double PXOVER = 0.8;
	final double PMUTATION = 0.1;
	double[] gene = new double [NVARS];
	double fitness;
	double[] upper = new double[NVARS];
	double[] lower = new double[NVARS];
	double rfitness;
	double cfitness;
	Genotype(){
		;
	}
}

File: OLDE_CPPF13/genotype.cpp
#include <cstring>
#include <iostream>
#include "test_func.cpp"
using namespace std;
class Genotype{
public:
	const static int NVARS = 30;
	const static double PXOVER = 0.5;
	const static double SCALE = 0.9;
	const static int LEVEL = 2;
	double gene[NVARS];
	double fitness;
	double upper[NVARS];
	double lower[NVARS];
	Genotype(){
		memset(gene,0,sizeof(gene));
		memset(upper,0,sizeof(upper));
		memset(lower,0,sizeof(lower));
		fitness = INF;
	}
	Genotype(Genotype &genotype){
		memcpy(this->gene,genotype.gene,sizeof(genotype.gene));	
		memcpy(this->upper,genotype.upper,sizeof(genotype.upper));
		memcpy(this->lower,genotype.lower,sizeof(genotype.lower));
		this->fitness = genotype.fitness;
	}
	~Genotype(){
	}
};
